python/plugins/processing/CMakeLists.txt

```diff
@@ -7,8 +7,10 @@ ADD_SUBDIRECTORY(images)
 ADD_SUBDIRECTORY(modeler)
 ADD_SUBDIRECTORY(script)
 ADD_SUBDIRECTORY(tools)
-ADD_SUBDIRECTORY(tests)
 ADD_SUBDIRECTORY(ui)
+IF (ENABLE_TESTS)
```

Comment:
Right - my reasoning is kind of abstract, so bear with me...

Firstly, from a software patterns POV, "plugins" should be self contained. One small example of this is that moving plugins in and out of core is simpler if the plugin carries and installs it's own tests.

And more abstractly, I've been thinking of plugins as python modules. Even though they aren't all currently built this way, I think we should move towards that. We're using python and the standard unit of application composition in Python is a module. The tests are part of the module and "/tests or /tests.py" is a standard place to store a modules tests.

Finally, I think core plugins should serve as an example for contributed plugins, and they should resemble contributed plugins as much as possible so that in core plugins we're dogfooding the same plugin tooling and practices that we want contributors to use. To my point - part of the plugin code is (or should be) tests, and I hope we can move more contributors to write tests for their plugin by making it easy to run plugin tests in situ. This plugin doesn't solve that, but I think moving Processing's tests out of its module would be a step in the wrong direction.